_posts/2025-04-25-invoking-component-functions-in-wasmtime-cli.md

```diff
@@ -0,0 +1,295 @@
+---
+title: "Running WebAssembly (Wasm) Components From the Command Line"
+author: "Tim McCallum"
+date: "2025-04-25"
+github_name: "tpmccallum"
+excerpt_separator: <!--end_excerpt-->
+---
+Wasmtime now supports invoking Wasm component exports directly from the command line with the new `--invoke` flag. 
+This article walks through building a Wasm component in Rust and using `wasmtime run --invoke` to execute specific functions (enabling powerful workflows for scripting, testing, and integrating Wasm into modern development pipelines).
+<!--end_excerpt-->
+
+## The Evolution of Wasmtime's CLI
+
+Wasmtime's `run` subcommand has traditionally supported running Wasm modules as well as invoking that **module**'s exported function. However, with the evolution of the Wasm Component Model, this article focuses on a newer capability; creating a component that exports a function and then demonstrating how to invoke that **component**'s exported function.
+
+By the end of this article, you’ll be ready to create Wasm components and orchestrate their exported component functions to improve your workflow's efficiency and promote reuse. Potential examples include:
+
+* Shell Scripting: Embed Wasm logic directly into Bash or Python scripts for seamless automation.
+* CI/CD Pipelines: Validate components in GitHub Actions, GitLab CI, or other automation tools without embedding them in host applications.
+* Cross-Language Testing: Quickly verify that interfaces match across implementations in Rust, JavaScript, and Python.
+* Debugging: Inspect exported functions during development with ease.
+* Microservices: Chain components in serverless workflows, such as compress → encrypt → upload, leveraging Wasm's modularity.
+
+## Tooling & Dependencies
+
+If you want to follow along, please install:
+
+* [Rust](https://www.rust-lang.org/tools/install) (if you already have Rust installed, make sure you are on [the latest version](https://github.com/rust-lang/rust/releases)),
+* [`cargo`](https://crates.io/crates/cargo) (if already installed, please make sure you are on [the latest version](https://crates.io/crates/cargo)),
+* [`cargo component`](https://crates.io/crates/cargo-component) (if already installed, please make sure you are on [the latest version](https://crates.io/crates/cargo-component)), and
+* [`wasmtime` CLI](https://docs.wasmtime.dev/cli-install.html) (or use a [precompiled binary](https://docs.wasmtime.dev/cli-install.html#download-precompiled-binaries)). If already installed, ensure you are using [the latest version](https://github.com/bytecodealliance/wasmtime/releases).
```

Comment:
We should wait to publish this until Wasmtime 33 is out - invoke didnt make it into the Wasmtime 32 release. I'd amend this to say it requires wasmtime 33 specifically, rather than the latest.

Reply:
Thank you @pchickey 
I have updated the filename and the `date` in the frontmatter to align with the next expected release.
I have also updated the `wasmtime` section to specify `v33.0.0` or newer.
Changes are in [cb41248](https://github.com/bytecodealliance/bytecodealliance.org/pull/126/commits/cb41248bddadd7721ec3e6e54cb72b0fee0c6fd3).
Ready for sign off cc:  @ricochet @ospencer @abrown @tschneidereit.
Kind regards
Tim

Reply:
@tpmccallum and I have talked about this but for anyone else wondering: this is on the agenda for the next TSC meeting (no hurry since we expect it to be published later in May).